core/vtk/ttkProjectionFromTable/ttkProjectionFromTable.cpp

```diff
@@ -0,0 +1,318 @@
+  const auto nTableValues = tableDataXArray->GetNumberOfTuples();
+
+  // --------------------------------------------------------------------------
+  // --- Get ordered values
+  // --------------------------------------------------------------------------
+  // store point index <-> ordering value in vector
+  std::vector<std::tuple<int, double, double>> orderedValues;
+
+#ifndef TTK_ENABLE_DOUBLE_TEMPLATING
+  switch(surfaceXArray->GetDataType()) {
+    vtkTemplateMacro(
+      dispatch(orderedValues,
+               static_cast<VTK_TT *>(ttkUtils::GetVoidPointer(surfaceXArray)),
+               static_cast<VTK_TT *>(ttkUtils::GetVoidPointer(surfaceYArray)),
+               nSurfaceValues));
+  }
+#else
+  switch(vtkTemplate2PackMacro(
+    surfaceXArray->GetDataType(), surfaceYArray->GetDataType())) {
+    vtkTemplate2Macro(
+      dispatch(orderedValues,
+               static_cast<VTK_T1 *>(ttkUtils::GetVoidPointer(surfaceXArray)),
+               static_cast<VTK_T2 *>(ttkUtils::GetVoidPointer(surfaceYArray)),
+               nSurfaceValues));
+  }
+#endif // TTK_ENABLE_DOUBLE_TEMPLATING
+
+  // Get number of unique values of each array
+  std::vector<double> xValues(orderedValues.size()),
+    yValues(orderedValues.size());
+  for(unsigned int i = 0; i < orderedValues.size(); ++i) {
+    auto tup = orderedValues[i];
+    xValues[i] = std::get<1>(tup);
+    yValues[i] = std::get<2>(tup);
+  }
+  std::sort(xValues.begin(), xValues.end());
+  const long nUniqueXValues
+    = std::unique(xValues.begin(), xValues.end()) - xValues.begin();
+  std::sort(yValues.begin(), yValues.end());
+  const long nUniqueYValues
+    = std::unique(yValues.begin(), yValues.end()) - yValues.begin();
+  std::array<const long, 2> surfaceDim{nUniqueXValues, nUniqueYValues};
+
+  if(nUniqueXValues * nUniqueYValues != surface->GetNumberOfPoints()) {
+    printErr(
+      "Number of unique values in first array times the number of unique "
+      "values in the second one does not equal the number of points");
+    return 0;
+  }
+
+  // Compare two pairs of index/value according to their values
+  double yRange[2] = {*std::min_element(yValues.begin(), yValues.end()),
+                      *std::max_element(yValues.begin(), yValues.end())};
+  double minXInterval = std::numeric_limits<double>::max();
+  for(unsigned int i = 1; i < nUniqueXValues; ++i)
+    minXInterval = std::min(minXInterval, xValues[i] - xValues[i - 1]);
+  const auto normValue = [&](double v) {
+    return (v - yRange[0]) / (yRange[1] - yRange[0]) * minXInterval * 0.99;
+  };
+  const auto cmp = [&](const std::tuple<int, double, double> &a,
+                       const std::tuple<int, double, double> &b) {
+    return std::get<1>(a) + normValue(std::get<2>(a))
+           < std::get<1>(b) + normValue(std::get<2>(b));
+  };
+
+  // sort the vector of indices/values in ascending order
+  std::sort(orderedValues.begin(), orderedValues.end(), cmp);
```

Comment:
this is a sequential sort.
please use OpenMp if available (see core/base/common/OrderDisambiguation.h:78, function sortVertices())

Reply:
Indeed, thanks. It is done in the next commit.
I've also modified PointSetToSurface and PointSetToCurve.